File: 2-variables-and-types/src/test/java/org/first/tutorial/ExerciseTest.java
package org.first.tutorial;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class ExerciseTest {

    @Test
    void calculatesInchesPerRotation() {
        Exercise exercise = new Exercise();
        exercise.wheelCount = 4;
        double inches = exercise.inchesPerWheelRotation(6.0);
        assertEquals(Math.PI * 6.0 * 4, inches, 0.0001);
    }

    @Test
    void formatsStatusLine() {
        Exercise exercise = new Exercise();
        exercise.driverName = "Mia";
        exercise.wheelCount = 4;
        exercise.batteryVoltage = 12.5;
        exercise.enabled = true;

        String status = exercise.statusLine();
        assertTrue(status.contains("Driver Mia"), "Driver name missing");
        assertTrue(status.contains("wheels:4"), "Wheel count missing");
        assertTrue(status.contains("battery:12.5V"), "Battery missing");
        assertTrue(status.contains("enabled:true"), "Enabled flag missing");
    }
}

File: kotlin/02-variables-and-types/src/test/kotlin/org/first/tutorial/ExerciseTest.kt
package org.first.tutorial

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ExerciseTest {

    @Test
    fun calculatesInchesPerRotation() {
        val exercise = Exercise()
        exercise.wheelCount = 4
        val inches = exercise.inchesPerWheelRotation(6.0)
        assertEquals(Math.PI * 6.0 * 4, inches, 0.0001)
    }

    @Test
    fun formatsStatusLine() {
        val exercise = Exercise()
        exercise.driverName = "Mia"
        exercise.wheelCount = 4
        exercise.batteryVoltage = 12.5
        exercise.enabled = true

        val status = exercise.statusLine()
        assertTrue(status.contains("Driver Mia"), "Driver name missing")
        assertTrue(status.contains("wheels:4"), "Wheel count missing")
        assertTrue(status.contains("battery:12.5V"), "Battery missing")
        assertTrue(status.contains("enabled:true"), "Enabled flag missing")
    }
}
